.github/workflows/storybook.yaml

```diff
@@ -1,19 +1,28 @@
-name: publish-storybook
+name: storybook
 env: 
     STORYBOOK_TOKEN: ${{ secrets.STORYBOOK_TOKEN }}
 on:
   push:
     branches:
       - master
+  pull_request:
+    branches:
+      - master
+      - 'release-*'      
 jobs:
-  publish:
-    if: github.repository_owner == 'rancher'
+  storybook:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - name: Use Node.js
       uses: actions/setup-node@v1
       with:
         node-version: '14.x'
-    - name: script
+    - name: Install packages
+      run: yarn install
+    - name: Build Storybook
+      run: |
+        yarn build-storybook
+    - name: Publish Storybook
```

Comment:
Could we cache the build and publish conditionally this one, so we could merge the 2 jobs into one? It would be easier to track.

Reply:
Yes but in the issue I've written that it would be better to have both the workflows in the same file.
In that way we will have the same steps and cannot risk to fail on `master`.
The publish part can be turned into another job, that's why we could cache it and pass to the other job. It will also help to understand where it does fail.

Alternatively we could add the condition for the step, but the name for the workflow/job would be confusing (`storybook/publish`)